docker/configure_workers_and_start.py

```diff
@@ -74,6 +74,9 @@
 MAIN_PROCESS_INSTANCE_NAME = "main"
 MAIN_PROCESS_LOCALHOST_ADDRESS = "127.0.0.1"
 MAIN_PROCESS_REPLICATION_PORT = 9093
+# Obviously, these would only be used with the UNIX socket option
+MAIN_PROCESS_UNIX_SOCKET_PUBLIC_PATH = "/run/mainpublic.sock"
```

Comment:
Seems like we should align this to the docs like `main_public.sock` for easier `Ctrl + f` finding

Reply:
This is part of the Complement testing infrastructure, if you would like that to be so, I will see that it is done.

Reply:
My preference is to align so it's just easier to find things and use it as a reference. For example, I used the Complement setup as a reference for the Matrix Public Archive tests

Reply:
Turns out this is slightly less organized than I would like. Can't(by default) create files in a *subdirectory* of `/var/run` such as I had it in the docs. The app-user would need create permissions, and I'm pretty sure they don't by default. Placing the sockets directly in `/var/run` works fine though. Is that secure enough? It's a little outside of my expertise on that matter. For docker images, it would be fine as it's all self contained. But I don't know about a debian or other such type of install.

Reply:
It's fine to keep it where it is but name it `main_public.sock`.

I don't have any insight into the security aspect of unix sockets

Reply:
Switched it around in [e3d4fb4](https://github.com/matrix-org/synapse/pull/15708/commits/e3d4fb41772411e463d90cb343ae53c2cb97043f)

Bonus included extra detail about not creating directories 'auto-magically' in [fe18b66](https://github.com/matrix-org/synapse/pull/15708/commits/fe18b66f3bcfbfcc2040bd7b47123b558366c559). This won't help with the security question, but might help them understand why something might break if they assign a socket file to a directory that doesn't exist.

Reply:
> might help them understand why something might break if they assign a socket file to a directory that doesn't exist.

What's the error when this occurs and where does it appear? Do we need to make it more clear?

Reply:
Looks clear enough :+1: